features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/APIDefinition/APIDefinition.jsx

```diff
@@ -0,0 +1,276 @@
+        try {
+            const result = JSON.parse(swagger);
+            return Object.prototype.toString.call(result) === '[object Object]'
+                || Array.isArray(result);
+        } catch (err) {
+            return false;
+        }
+    }
+
+    /**
+     * Handles the file upload.
+     * */
+    onDrop(file) {
+        if (file[0]) {
+            let reader = new FileReader();
+            reader.onload = (e) => {
+                let content = e.target.result;
+                this.setState({swagger: content}, () => {
+                    this.updateSwaggerDefinition()
+                });
+            }
+            reader.readAsText(file[0]);
+        } else {
+            Alert.error('Unsupported file type.')
+        }
+    }
+
+    render() {
+        const {swagger} = this.state;
+        const {classes} = this.props;
+
+        const editorOptions = {
+            selectOnLineNumbers: true,
+            readOnly: true,
+            smoothScrolling: true,
+            wordWrap: 'on',
+            handleMouseWheel: true,
+            scrollbar: {
+                handleMouseWheel: true
+            }
+        };
+        if (this.state.notFound) {
+            return <ResourceNotFound message={this.props.resourceNotFountMessage}/>;
+        }
+        if (!swagger) {
+            <Progress />
+        }
+
+        return (
+            <div className={classes.root}>
+                <div className={classes.titleWrapper}>
+                    <Typography variant='h4' align='left' className={classes.mainTitle}>
+                        API Definition
+                    </Typography>
+                    <Button size='small' className={classes.button} onClick={this.openEditor}>
+                        <EditRounded className={classes.buttonIcon}/>
+                        Edit
+                    </Button>
+                    <Dropzone
+                        multiple={false}
+                        className={classes.dropzone}
+                        accept={['application/json', 'application/x-yaml']}
+                        onDrop={(dropFile) => {
+                            this.onDrop(dropFile);
+                        }}>
+                        <Button size='small' className={classes.button}>
+                            <CloudUploadRounded className={classes.buttonIcon}/>
+                            Import API Definition
+                        </Button>
+                    </Dropzone>
+                </div>
+                <div>
+                    <MonacoEditor
+                        width='100%'
+                        height='calc(100vh - 51px)'
+                        theme='vs-dark'
+                        value={swagger}
+                        options={editorOptions}
+                    />
+                </div>
+                <Dialog fullScreen open={this.state.openEditor} onClose={this.closeEditor}
+                        TransitionComponent={this.Transition}>
```

Comment:
instead of passing an instance method as stateless components, shall we call the function and return required elements from it, so we could use lowercase method name there,

Reply:
fixed https://github.com/wso2/carbon-apimgt/pull/6572/commits/547554b9a9c260c3115ce9cd4e74ad8008c3d85a